docs/hacking-an-agent.md

```diff
@@ -43,39 +43,41 @@ from .conftest import CUR_PATH, DummyConnection
 
 ## Initialise the agent
 
-Create a node.
-
+Create a `Connection` and a `MailBox`.
 ``` python
-node = LocalNode()
+stub_connection = StubConnection(input_file_path='input.txt', output_file_path='output.txt')
+mailbox = MailBox(stub_connection)
 ```
 
-Initialise a `MailBox` object with a public key and grab the private key and add it to a wallet.
-
+Create a wallet with a private key.
 ``` python
-public_key_1 = "mailbox1"
-mailbox1 = MailBox(OEFLocalConnection(public_key_1, node))
 private_key_pem_path = os.path.join(CUR_PATH, "data", "priv.pem")
 wallet = Wallet({'default': private_key_pem_path})
 ```
 
-Using a variable for accessing running ledgers, initialise the agent. 
+Get the public key from the wallet. You would use this to talk to an OEF or Fetch.ai ledger node.
```

Comment:
I would comment this out. We might use it another time but here it does not add value.

Reply:
I put the pk comments in a Note. Awaiting your approval.